spanner/integration_test.go

```diff
@@ -390,12 +405,17 @@ loop:
 
 // Test SingleUse transaction.
 func TestIntegration_SingleUse(t *testing.T) {
+	skipEmulatorTestForPG(t)
 	t.Parallel()
 
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 	defer cancel()
-	// Set up testing environment.
-	client, _, cleanup := prepareIntegrationTest(ctx, t, DefaultSessionPoolConfig, singerDBStatements)
+
+	statements := singerDBStatements
```

Comment:
I assume that in the end most (all?) tests will be executed using both dialects. In that case I think it makes more sense to add an extra parameter to `prepareIntegrationTest` instead of this `if` statement. When there is only one or maybe a few tests that also run on PG then this makes sense, but as soon as the majority is tested against both dialects, these `if` statements will be in every test.

Reply:
I have refactored the statement selection code using map based on dialect, wdyt?
cc: @olavloite @hengfengli

Reply:
That seems like a good strategy to me.